### PR TITLE
Incorrect unlock times

### LINKED ISSUE #5049: Incorrect unlock times
### Expected behavior
The unlock times for selfstake and a regular stake should be correct
### Actual behavior
image
Could be a service issue
### Steps to reproduce
Stake/Unstake and check the locktimes
### Which version(s) does this affect? (Environment, OS, etc...)
Desktop 3.0-beta

### PR DESCRIPTION
**Pull request by oskarleonard**
### What was the problem?

This PR resolves #5049

### How was it solved?

Use the `expectedUnlockTime` field from BE to determine when a stake can be unlocked

### How to test?

1. Stake 100 with a validator
2. Edit that stake to 80
3. Go to http://0.0.0.0:8080/#/validators/profile/stakes?modal=lockedBalance
4. You should see a text with "20 LSK" - "will be available to unlock in a month"

### CODE REVIEW

**Review by ManuGowda — COMMENTED** [review]
Screenshot 2023-06-14 at 7 48 16 AM

src/modules/pos/validator/components/StakeStatus/statusMessages.js

This is part of the logic that needs refactoring.

File: src/modules/pos/validator/components/StakeStatus/statusMessages.js [hunk 1]
```diff
@@ -23,13 +21,13 @@ const getSuccessMessage = (t, locked, unlockable, selfUnstake = { confirmed: 0 }
         {regularUnlockable > BigInt(0) ? (
           <>
             <LiskAmountFormatted val={regularUnlockable.toString()} token={token} />{' '}
-            <span>{t('will be available to unlock in {{unlockTime}}h.', { unlockTime })}</span>
+            <span>{t('Your tokens will be available to unlock when the locking period ends')}</span>
```
Comment (ManuGowda) [inline src/modules/pos/validator/components/StakeStatus/statusMessages.js]: when the locking period ends.

File: src/modules/pos/validator/components/StakeStatus/statusMessages.js [hunk 2]
```diff
@@ -23,13 +21,13 @@ const getSuccessMessage = (t, locked, unlockable, selfUnstake = { confirmed: 0 }
...
           </>
         ) : null}
         {selfUnstakeUnlockable > 0 ? (
           <>
             <LiskAmountFormatted val={selfUnstakeUnlockable} token={token} />{' '}
-            <span>{t('will be available to unlock in 1 month.')}</span>
+            <span>{t('Your tokens will be available to unlock when the locking period ends')}</span>
```
Comment (ManuGowda) [inline src/modules/pos/validator/components/StakeStatus/statusMessages.js]: when the locking period ends.

File: src/modules/pos/validator/components/StakeStatus/statusMessages.js [hunk 3]
```diff
@@ -49,7 +47,7 @@ const getSuccessMessage = (t, locked, unlockable, selfUnstake = { confirmed: 0 }
         <span>{t('You have now locked')}</span> <LiskAmountFormatted val={locked} token={token} />{' '}
         <span>{t('for staking and may unlock')}</span>{' '}
         <LiskAmountFormatted val={unlockable} token={token} />{' '}
-        <span>{t('in {{unlockTime}} hours.', { unlockTime })}</span>
+        <span>{t('when the locking period ends,')}</span>
```
Comment (ManuGowda) [inline src/modules/pos/validator/components/StakeStatus/statusMessages.js]: when the locking period ends.

**Review by ManuGowda — CHANGES_REQUESTED** [review]
Screenshot 2023-06-14 at 4 01 59 PM

Need to handle the invalid date case

**Review by ManuGowda — APPROVED** [review]

**Review by ikem-legend — APPROVED** [review]

**Pull request merged**